From 2eb230e185c59144e987232eaf8e6e2a16007f05 Mon Sep 17 00:00:00 2001
From: "Wang, ArvinX" <arvinx.wang@intel.com>
Date: Tue, 22 May 2018 19:18:24 +0800
Subject: [PATCH 1/2] Modify the LensResolverActivity intent with right
 component

The music can't play due to the LensResolverActivity always
use the default intent to start activity, correct it using the
ResolveInfo component.

Change-Id: Id760c0e94857cda8659b3f3bcc5ae8359370884e
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-63290
Signed-off-by: Wang, ArvinX <arvinx.wang@intel.com>
---
 .../android/support/car/lenspicker/LensResolverActivity.java   | 10 ++++++++--
 src/com/android/support/car/lenspicker/ResolverAdapter.java    |  4 +---
 2 files changed, 9 insertions(+), 5 deletions(-)

diff --git a/src/com/android/support/car/lenspicker/LensResolverActivity.java b/src/com/android/support/car/lenspicker/LensResolverActivity.java
index 6b0892a..3c8103f 100644
--- a/src/com/android/support/car/lenspicker/LensResolverActivity.java
+++ b/src/com/android/support/car/lenspicker/LensResolverActivity.java
@@ -164,8 +164,14 @@ public class LensResolverActivity extends Activity implements
      * {@link #mResolveIntent}.
      */
     @Override
-    public void onActivitySelected(ResolveInfo info, LensPickerItem item) {
-        ComponentName component = item.getLaunchIntent().getComponent();
+    public void onActivitySelected(ResolveInfo info, LensPickerItem item /*not use*/) {
+        ComponentName component = info.getComponentInfo().getComponentName();
+
+        if (LensPickerUtils.isMediaService(info)) {
+            Intent intent = LensPickerUtils.getMediaLaunchIntent(getPackageManager(), LensPickerUtils.getPackageName(info),
+                    info.serviceInfo.name);
+            component = intent.getComponent();
+        }
 
         if (mAlwaysCheckbox.isChecked()) {
             IntentFilter filter = buildIntentFilterForResolveInfo(info);
diff --git a/src/com/android/support/car/lenspicker/ResolverAdapter.java b/src/com/android/support/car/lenspicker/ResolverAdapter.java
index b7fb8e2..00ea2f9 100644
--- a/src/com/android/support/car/lenspicker/ResolverAdapter.java
+++ b/src/com/android/support/car/lenspicker/ResolverAdapter.java
@@ -93,8 +93,6 @@ public class ResolverAdapter extends RecyclerView.Adapter<ResolverListRow>
 
             for (ResolveInfo info : mResolveInfos) {
                 String packageName = LensPickerUtils.getPackageName(info);
-                Intent launchIntent = LensPickerUtils.getLaunchIntent(packageName, info,
-                        packageManager);
 
                 try {
                     ApplicationInfo aInfo = packageManager.getApplicationInfo(packageName, 0);
@@ -116,7 +114,7 @@ public class ResolverAdapter extends RecyclerView.Adapter<ResolverListRow>
                     }
 
                     items.add(new LensPickerItem(displayName,
-                            getComponentIcon(packageManager, aInfo), launchIntent,
+                            getComponentIcon(packageManager, aInfo), null /* launchIntent */,
                             null /* facetId */));
                 } catch (PackageManager.NameNotFoundException e) {
                     // skip this package.
-- 
1.9.1

